source/browseMode.py

```diff
@@ -477,10 +488,51 @@ def _activatePosition(self,obj=None):
 			self._activateNVDAObject(obj)
 
 	def script_activatePosition(self,gesture):
-		self._activatePosition()
+		self.maybeSyncFocus(postFocusFunc=self._activatePosition)
 	# Translators: the description for the activatePosition script on browseMode documents.
 	script_activatePosition.__doc__ = _("Activates the current object in the document")
 
+	def maybeSyncFocus(self, postFocusFunc=None):
+		if  config.conf["virtualBuffers"]["focusFollowsBrowse"]:
+			if postFocusFunc is not None:
+				postFocusFunc()
+			return
+		try:
+			obj = self._lastFocusableObject
+		except AttributeError:
+			return
+		if not obj:
+			return
+		synchronousCall =True
+		setFocusCall = False 
+		if obj!=self.rootNVDAObject and self._shouldSetFocusToObj(obj) and obj!= api.getFocusObject():
+			setFocusCall = True
+			synchronousCall = False
+		if obj.hasFocus:
+			synchronousCall = True
+
+		if not synchronousCall:
+			uid = getObjectID(obj)
+			if uid in self.postFocusFuncDict:
+				log.error("postFocusFunc has not been called asynchronously")
+			if postFocusFunc is not None:
+				self.postFocusFuncDict[uid] = postFocusFunc
+		if setFocusCall:
+			obj.setFocus()
+		if synchronousCall:
+			if postFocusFunc is not None:
+				postFocusFunc()
+
+	def maybeSyncFocusAndPassThrough(self, gesture=None): 		
+		self.maybeSyncFocus()
+		if gesture:
+			gesture.send()

```

Comment:
I'm not sure if this method is necessary. I think it is more readable to keep the gesture.send in the original places outside of this method.